Added new file to extract output

2001asjad opened the pull request:
Added new file to extract the Quoted contents.
Edited the `collect_data.py` file to add new `issue_content_quotation_path` to the MongoDB database

Signed-off-by: 2001asjad <iitasjad2001@gmail.com>

2001asjad commented:
@LongyuZhang @llxia Can you please take a look at the addressed changes.
Thank You

LongyuZhang commented:
@2001asjad Could you run the script for a few minutes, and paste some examples here, in pairs of "original GitHub issue content" and "extracted quotation content"? Thanks.

2001asjad commented:
Adding the screenshots in pairs of original issue content and the extracted ones:-

**Pair 1**
Original:
![aqa-tests_467](https://user-images.githubusercontent.com/60435499/122258110-4f207e80-ceee-11eb-84d5-0deb4f088291.PNG)
Extracted:
![extracted_467](https://user-images.githubusercontent.com/60435499/122258196-65c6d580-ceee-11eb-820d-5d8b7bb4b9db.PNG)

**Pair 2**
Original:
![aqa-tests_665](https://user-images.githubusercontent.com/60435499/122258264-7c6d2c80-ceee-11eb-991d-6b583fd4afda.PNG)
Extracted:
![extracted_665](https://user-images.githubusercontent.com/60435499/122258316-898a1b80-ceee-11eb-863a-ab87746d98a8.PNG)

**Pair 3**
Original:
![aqa-tests_795](https://user-images.githubusercontent.com/60435499/122258429-a6265380-ceee-11eb-80ee-00cc128efa24.PNG)
Extracted:
![extracted_795](https://user-images.githubusercontent.com/60435499/122258465-b0485200-ceee-11eb-8e4c-760fec40dc8c.PNG)

The original extracted content is saved at `./data/GitHubData/IssueContentQuote/` folder.
Only those contents that have the quoted output are extracted and added to the new folder.

2001asjad commented:
@LongyuZhang Waiting for your suggestions.
Thank You

LongyuZhang commented on MachineLearningPrototype/dataCollection/collect_data.py:
```diff
@@ -6,6 +6,7 @@
 import time
 from pprint import pprint
 import pymongo
+from extract_jenkins import *
```
Could you use function name to replace `*`?

LongyuZhang reviewed (commented):
The sample looks great! 
Another suggestion is to put the function in `extract_jenkins.py` into [utils/preprocess_data](https://github.com/adoptium/aqa-test-tools/blob/master/MachineLearningPrototype/utils/preprocess_data.py) directly, so we can also utilize this function in other cases.
@llxia Would you have other suggestions on this PR? Thanks.

2001asjad commented:
> The sample looks great!
> Another suggestion is to put the function in `extract_jenkins.py` into [utils/preprocess_data](https://github.com/adoptium/aqa-test-tools/blob/master/MachineLearningPrototype/utils/preprocess_data.py) directly, so we can also utilize this function in other cases.
> @llxia Would you have other suggestions on this PR? Thanks.

For sure @LongyuZhang .  Gonna do it and update you soon

2001asjad commented:
@LongyuZhang @llxia  please take a look. I have deleted the `extract_jenkins.py` file and copied it's function into the `preprocess_data.py` file.
Waiting for the suggestions.
Thank You

llxia commented on MachineLearningPrototype/dataCollection/collect_data.py:
```diff
@@ -19,16 +22,21 @@ def store_on_fs(data, file_name):
+			with open(file_name, "w") as f:
+				f.write(str(data))
+		except:
+			return
```
Can we print the exception?

2001asjad replied:
@llxia it would be great if you could explain me this a bit more.
Like if there aren't any quotations in a particular file, it stores a `[]` in the new file that would be created in the IssueContentQuote directory.
Are you talking about this or something else?
Thanks

LongyuZhang commented on MachineLearningPrototype/utils/preprocess_data.py:
```diff
@@ -8,3 +8,9 @@ def remove_time_stamp(content):
+def extract_quotation_content(content):
+    pattern = "\`\`\`([\s\S]*?)\`\`\`"
+    substring = re.findall(pattern, content)
+    result=''.join(substring)
```
Instead of empty string '', we can use string with a space (' '), in case two quoted contents do not have line change or space in between.

2001asjad commented:
@LongyuZhang addressed the changes, please take a look.
Thanks.

LongyuZhang commented on MachineLearningPrototype/utils/preprocess_data.py:
```diff
@@ -8,3 +8,9 @@ def remove_time_stamp(content):
+    pattern = "\`\`\`([\s\S]*?)\`\`\`"
+    substring = re.findall(pattern, content)
+    result=' '.join(substring)
+    return result
```
- Condition check may be needed, e..g `if (len(substrings) < 2)`, return substrings, else join substrings and return. 
- Could you also address Lan's comment of printing exception info? Thanks.

2001asjad commented:
@LongyuZhang @llxia I have made the changes, can you please take a look.
Thank You so much

LongyuZhang reviewed (commented):
@2001asjad The PR looks great now. Just need to adjust the format as mentioned below.
Also, could you run the script locally to test, and share some results? Thanks.

2001asjad commented:
Hello @LongyuZhang @llxia I am attaching the Screenshots to define the current scenario.
### Original FIle and Extracted file:
![aqa467](https://user-images.githubusercontent.com/60435499/122632840-f66e0300-d0f2-11eb-97f6-56b794026f6a.PNG)
![Extracted467](https://user-images.githubusercontent.com/60435499/122632843-fb32b700-d0f2-11eb-8494-a73682653bde.PNG)
### How the file structure would look:
![Snip1](https://user-images.githubusercontent.com/60435499/122632867-10a7e100-d0f3-11eb-8f89-73eedcba44d2.PNG)
![Snip2](https://user-images.githubusercontent.com/60435499/122632869-156c9500-d0f3-11eb-9d39-2ed0936a1d17.PNG)
Notice only those files get copied into the `IssueContentQuote` folder that have the quoted outputs.
### The Demo of the MongoDB:
![DBNew](https://user-images.githubusercontent.com/60435499/122632911-519ff580-d0f3-11eb-8ecb-c6a63dbdd3f7.PNG)

The only glitch we have as of now is that the files that aren't in the `IssueContentQuote` but in the `IssueContent` folder also get added in the `issue_content_quotation_path` of the MongoDB.
Looking forward for your reviews.
Thank You so much

LongyuZhang reviewed (approved):
LGTM. Thanks @2001asjad for working on this!
